### PR TITLE
[Enhancement]aggregation pipeline shared limit

### PR DESCRIPTION
**Pull request by zombee0**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/pipeline/aggregate/aggregate_blocking_sink_operator.h [hunk 1]
```diff
@@ -58,22 +58,28 @@ class AggregateBlockingSinkOperator : public Operator {
...
 };
 
 class AggregateBlockingSinkOperatorFactory final : public OperatorFactory {
 public:
     AggregateBlockingSinkOperatorFactory(int32_t id, int32_t plan_node_id, AggregatorFactoryPtr aggregator_factory,
                                          const SpillProcessChannelFactoryPtr& _)
             : OperatorFactory(id, "aggregate_blocking_sink", plan_node_id),
-              _aggregator_factory(std::move(aggregator_factory)) {}
+              _aggregator_factory(std::move(aggregator_factory)) {
+        _runtime_limit.store(_aggregator_factory->aggregator_param()->limit);
```
Comment (satanson) [inline be/src/exec/pipeline/aggregate/aggregate_blocking_sink_operator.h:70]: suggest to rename _runtime_limit to _hash_size_countdown?

File: fe/fe-core/src/main/java/com/starrocks/planner/AggregationNode.java [hunk 2]
```diff
@@ -270,6 +270,7 @@ protected void toThrift(TPlanNode msg) {
         msg.agg_node.setAgg_func_set_version(FeConstants.AGG_FUNC_VERSION);
         msg.agg_node.setInterpolate_passthrough(
                 useStreamingPreagg && ConnectContext.get().getSessionVariable().isInterpolatePassthrough());
+        msg.agg_node.setEnable_runtime_limit(ConnectContext.get().getSessionVariable().getEnableAggregationRuntimeLimit());
```
Comment (satanson) [inline fe/fe-core/src/main/java/com/starrocks/planner/AggregationNode.java:273]: can we use another name, shared_limit?

File: be/src/exec/pipeline/aggregate/aggregate_blocking_sink_operator.h [hunk 3]
```diff
@@ -58,22 +58,28 @@ class AggregateBlockingSinkOperator : public Operator {
     std::atomic_bool _is_finished = false;
     // whether enable aggregate group by limit optimize
     bool _agg_group_by_with_limit = false;
+    std::atomic<int64_t>* _runtime_limit = nullptr;
```
Comment (satanson) [inline be/src/exec/pipeline/aggregate/aggregate_blocking_sink_operator.h:61]: suggest to use std::atomic<int64_t>& type, then initialize it in ctor.
distinct_blocking_sink_operator also needs this optimization.

Comment (zombee0) [reply be/src/exec/pipeline/aggregate/aggregate_blocking_sink_operator.h:61]: Yes, correct, will do it.

File: fe/fe-core/src/main/java/com/starrocks/planner/AggregationNode.java (hunk 2 above)
Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/planner/AggregationNode.java:273]: ok, pipeline_shared_limit?

**Review by stdpain — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Review by DorianZheng — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by zombee0** [comment]
https://github.com/Mergifyio backport branch-3.2

**Comment by zombee0** [comment]
https://github.com/Mergifyio backport branch-3.3

**Pull request merged**